### ISSUE TITLE
Image Overlay Notifications Have Stopped Working

### ISSUE THREAD

**Issue by eatticks** [ascott18/TellMeWhen#2193]
### WoW Version

Retail

### TellMeWhen Version

11.0.4

### Describe the bug

My image overlay icons have stopped working (i.e., the image that is supposed to replace the base one when the condition is satisfied does not appear).  

Example:  for my enhance shaman, I have a MSW stack counting icon that tracks stacks of MSW buff and shows MSW pic by default and I have notifications to change the pic if certain things occur (if i get a stormweaver buff, pic changes to heal pic, etc.). Now, the normal MSW buff icon just stays the whole time and other pictures don't appear when they should.

They have always worked historically.  Thanks!

**Comment by eatticks** [comment]
^1^T^SShowTimer^B ^SType^Sbuff ^SName^SMaelstrom~`Weapon ^SEvents^T ^N1^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SPeriod^N120 ^SImage^S8004 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^SStormweaver ^t^N2^T ^SType^SBUFFDUR ^SOperator^S> ^SAndOr^SOR ^SName^SSurging~`Currents ^t^Sn^N2 ^t^t^N2^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N120 ^SImage^S454015 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^S454015 ^t^Sn^N1 ^t^t^N3^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N120 ^SImage^S51533 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SSPELLCD ^SName^SFeral~`Spirit ^t^N2^T ^SType^SPTSINTAL ^SLevel^N1 ^SName^SElemental~`Spirits ^t^N3^T ^SType^SCOMBAT ^t^Sn^N3 ^t^t^N4^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N15 ^SImage^S342240 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^SHailstorm ^t^N2^T ^SType^SBUFFDUR ^SName^SIce~`Strike ^t^Sn^N2 ^t^t^N5^T ^SType^SAnimations ^SOnlyShown^B ^SImage^S410681 ^SAnimation^SACTVTNGLOW ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFSTACKS ^SOperator^S> ^SName^SMaelstrom~`Weapon ^t^N2^T ^SType^SSPELLCD ^SAndOr^SOR ^SName^SPrimordial~`Wave ^t^N3^T ^SType^SPTSINTAL ^SLevel^N1 ^SName^SPrimordial~`Wave ^t^Sn^N3 ^t^t^N6^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N30 ^SImage^S334195 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^SHailstorm ^t^N2^T ^SType^SSPELLCD ^SName^SFrost~`Shock ^t^Sn^N2 ^t^t^N7^T ^SType^SAnimations ^SOnlyShown^B ^SMagnitude^N2 ^SImage^S410681 ^SAnimation^SICONSHAKE ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFSTACKS ^SOperator^S>= ^SLevel^N5 ^SName^SMaelstrom~`Weapon ^t^N2^T ^SType^SICON ^SAndOr^SOR ^SIcon^STMW:icon:1bRtx2wVn1eL ^t^Sn^N2 ^t^t^N8^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SPeriod^N30 ^SImage^S164887 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SICON ^SIcon^STMW:icon:1Z8l4XcKNCak ^t^N2^T ^SType^SPVPTALENTLEARNED ^SLevel^N1 ^SName^SStormweaver ^t^Sn^N2 ^t^t^N9^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N30 ^SImage^S8004 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SHEALTH ^SOperator^S<= ^SLevel^N60 ^t^N2^T ^SType^SPVPTALENTLEARNED ^SLevel^N1 ^SName^SStormweaver ^t^Sn^N2 ^t^t^N10^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N600 ^SImage^S375986 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SSPELLCD ^SName^SPrimordial~`Wave ^t^N2^T ^SType^SPTSINTAL ^SLevel^N1 ^SName^SPrimordial~`Wave ^t^N3^T ^SType^SPTSINTAL ^SName^SPrimal~`Maelstrom ^t^Sn^N3 ^t^t^N11^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N30 ^SImage^S188389 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^SPrimordial~`Wave ^t^N2^T ^SType^SDEBUFFDUR ^SUnit^Starget ^SName^SFlame~`Shock ^t^N3^T ^SType^SSPELLCD ^SName^SFlame~`Shock ^t^Sn^N3 ^t^t^N12^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N300 ^SImage^S327163 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SSPELLCD ^SName^SPrimordial~`Wave ^t^N2^T ^SType^SREACT ^SUnit^Starget ^SLevel^N1 ^t^N3^T ^SType^SPTSINTAL ^SLevel^N1 ^SName^SPrimordial~`Wave ^t^N4^T ^SType^SBUFFSTACKS ^SOperator^S< ^SLevel^N5 ^SName^SMaelstrom~`Weapon ^t^Sn^N4 ^t^SMagnitude^N2 ^t^N13^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N120 ^SImage^S191634 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SBUFFDUR ^SOperator^S> ^SName^SPrimordial~`Wave ^t^N2^T ^SType^SDEBUFFDUR ^SOperator^S> ^SUnit^Starget ^SName^SFlame~`Shock ^t^Sn^N2 ^t^t^N14^T ^SFade^b ^SType^SAnimations ^SAlpha^N1 ^SOnlyShown^B ^SPeriod^N500.1 ^SImage^S117014 ^SAnimation^SICONOVERLAYIMG ^SEvent^SWCSP ^SOnConditionConditions^T ^N1^T ^SType^SSPELLCHARGES ^SOperator^S> ^SName^SElemental~`Blast ^t^N2^T ^SType^SPTSINTAL ^SLevel^N1 ^SName^SElemental~`Blast ^t^N3^T ^SType^SCOMBAT ^t^N4^T ^SType^SBUFFSTACKS ^SOperator^S>= ^SLevel^N5 ^SName^SMaelstrom~`Weapon ^t^Sn^N4 ^t^t^N15^T ^SType^SSound ^SSound^SElectrical~`Spark ^SEvent^SOnCondition ^SOnConditionConditions^T ^N1^T ^SType^SBUFFSTACKS ^SLevel^N10 ^SName^SMaelstrom~`Weapon ^t^Sn^N1 ^t^t^Sn^N15 ^t^SSettingsPerView^T ^Sicon^T ^STextLayout^STMW:textlayout:1Z9zqgi9jRqi ^STexts^T ^N1^S[Stacks:Hide(0)] ^N2^S[Stacks:Hide(0)] ^t^t^t^SStates^T ^N2^T ^SAlpha^N0.5 ^t^t^SEnabled^B ^SGUID^STMW:icon:1Z61dVCx3XC3 ^t^N110401^S~`~| ^Sicon^^

^1 ^T^N1^T ^SName^S2002~`Bold ^SSize^N16 ^t^SGUID^STMW:textlayout:1Z9zqgi9jRqi ^SName^SCenter~`Number~`Custom~`2 ^t^N110401^S~`~| ^Stextlayout^STMW:textlayout:1Z9zqgi9jRqi ^^

**Comment by ascott18** [comment]
Unable to reproduce. 

Please do not ignore the issue template that is provided when you create a new issue. The questions and sections within it provide me with essential information to be able to diagnose and resolve your issue. If you do not include this information, there is little I can do to help you.

Please post the export string(s) of the icons, groups, or profiles you're having trouble with. 

To get an export string, open the icon editor, and click the button labeled "Import/Export/Backup". Select the "To String" option for the appropriate export type (icon, group, or profile), and then press CTRL+C to copy it to your clipboard.

Additionally, if applicable, add screenshots to help explain your problem. You can paste images directly into GitHub issues, or you can upload files as well.

**Comment by eatticks** [comment]
I think I figured out what is going on.  If you set a time for the notification icon to last, it usually starts immediately when the condition is satisfied.  What seems to be happening is that, once the condition is met, the default icon remains until the flash period expires at which point the icon pops up.

So, for example, if I have MSW as my default icon and I set lava burst to show for 15 seconds when it is off CD, then historically it would show for 15 seconds as soon as it is ready.  Now, once lava burst is ready, the default icon goes first for 15 seconds and THEN lava burst shows 15 seconds later. 

I think the ordering got mixed up somehow so that notification icons of fixed duration have to wait until the default icon gets a turn first for that amount of time which isn't how it should work (since it defeats the purpose of the notification). 

If i have this right, can this please get fixed so that the notification icon pops up as soon as the condition is met?  

Thanks!
